server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/mdc/LoggingResponseFilter.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.mdc;
+
+import org.apache.james.util.MDCStructuredLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class LoggingResponseFilter implements Filter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingResponseFilter.class);
+    private static final String STATUS = "status";
+    private static final String RESPONSE_BODY = "response-body";
+
+    @Override
+    public void handle(Request request, Response response) throws Exception {
+        MDCStructuredLogger.forLogger(LOGGER)
+            .addField(STATUS, response.status())
+            .addField(RESPONSE_BODY, request.body())
```

Comment:
I guess you mean response.body() ?
So given this, no unit test?

Reply:
We don't test logging so far. Should I?

Reply:
anyway you need to fix this, no?
about testing I don't know, maybe just testing that the handle method is filling the fields as expected can be fine if it prevents from such mistakes

Reply:
done